code/modules/mob/dead/observer/verbs.dm

```diff
@@ -519,6 +519,35 @@
 	else
 		to_chat(src, "<span class='warning'>Unable to find any MoMMI Spawners ready to build a MoMMI in the universe. Please try again.</span>")
 
+/mob/dead/observer/verb/become_hobo()
+	set name = "Become Space Hobo"
+	set category = "Ghost"
+	if(!can_reenter_corpse)
+		to_chat(src, "<span class='warning'>You must have had presence on this plane to become this.</span>")
+		return
+	if(!config.respawn_as_hobo)
+		to_chat(src, "<span class='warning'>Respawning as Space Hobo is disabled.</span>")
+		return
+
+	var/timedifference = world.time - client.time_died_as_mouse
+	if(client.time_died_as_mouse && timedifference <= mouse_respawn_time * 600)
+		var/timedifference_text
+		timedifference_text = time2text(mouse_respawn_time * 600 - timedifference,"mm:ss")
+		to_chat(src, "<span class='warning'>You may only spawn again as this role more than [mouse_respawn_time] minutes after your death. You have [timedifference_text] left.</span>")
+		return
+
+	if(!hobostart.len || hobostart.len < 9)
+		generate_hoboshack()
```

Comment:
this is odd because it means shacks can generate after round start... meaning a miner could walk past a section of roid and see it is empty, then come back later and a shack has appeared

I guess it's fine

Reply:
Intended, but if you have any better ideas, I'd like to know

Reply:
You can generate a reasonable number of shacks (2-3) at shift start. Since you don't seem to mind spawning in hobos at already-populated shacks, this is probably enough shacks. Besides, they will probably get immediately abandoned.

Reply:
That was the original idea, I'll see what I can do

Reply:
That was the original idea, done